hexadecimal/solution.py

```diff
@@ -1,10 +1,17 @@
-from functools import reduce
+_hex_code = dict(zip('0123456789', range(10)))
+_hex_code.update(zip('abcdef', range(10, 16)))
+_hex_code.update(zip('ABCDEFG', range(10, 16)))
 
 
 def hexa(s):
-    s = s.lower()
-    if set(s) - set('0123456789abcdef'):
+    result = 0
+    c = None
+    try:
+        for c in s:
+            result = result*16 + _hex_code[c]
+    except KeyError:
+        c = None
```

Comment:
Maybe it would be more transparent if there would be first a check if the length of the input was > 0. Then a `ValueError` could be raised here directly. What do you think?

Reply:
This solution is focusing on EAFP instead of LBYL intentionally. I think if you want to merge it as solution2.py, then that is a nice contrast to the original solution and people can explore both startegies. I'll add a comment explaining the idea.

Reply:
Sure, that's fine.